apps/web/global/GlobalErrorBoundary.tsx

```diff
@@ -0,0 +1,28 @@
+"use client"
+import { Component, ErrorInfo, ReactNode } from "react"
+import { toast } from "./components/ui/use-toast"
+
+type Props = {
+  toast: typeof toast
+  children: ReactNode
+}
+type State = {
+  hasError: boolean
+}
+
+export class GlobalErrorBoundary extends Component<Props, State> {
+  constructor(props: Props){
+    super(props)
+  }
+  componentDidCatch(_error: Error, _errorInfo: ErrorInfo): void {
+    this.props.toast({
+      title: "Oops",
+      variant: "destructive",
+      description: 'Algo deu errado 🥶',
+    });
+  }
+
+  render() {
+    return this.props.children;
+  }
+}
```

Comment:
Is there a way to do this with react hooks?

Also, as we're using next, maybe we could use the suggested error handler https://nextjs.org/docs/app/building-your-application/routing/error-handling

You could try this by throwing an error on the click of the button of the Hero section